src/TLuaInterpreter.cpp

```diff
@@ -7041,48 +7041,60 @@ int TLuaInterpreter::invokeFileDialog(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getTimestamp
 int TLuaInterpreter::getTimestamp(lua_State* L)
 {
-    int luaLine;
-    int args = lua_gettop(L);
-    int n = 1;
-    string name = "";
-    if (args < 1) {
-        return 0;
-    }
-    if (args == 2) {
-        if (lua_isstring(L, n)) {
-            name = lua_tostring(L, n);
-            n++;
+    int n = lua_gettop(L);
+    int s = 0;
+    QString name;
+    if (n > 1) {
+        if (!lua_isstring(L, ++s)) {
+            lua_pushfstring(L, "getTimestamp: bad argument #%d type (mini console, user window or buffer name as string expected {may be omitted for the \"main\" console}, got %s!)", s, luaL_typename(L, s));
+            return lua_error(L);
+        } else {
+            name = QString::fromUtf8(lua_tostring(L, s));
+            if (name == QLatin1String("main")) {
+                // clear it so it is treated as the main console below
+                name.clear();
+            }
         }
     }
 
-    if (!lua_isnumber(L, n)) {
-        lua_pushstring(L, "getTimestamp: wrong argument type");
-        lua_error(L);
-        return 1;
+    qint64 luaLine;
+    if (!lua_isnumber(L, ++s)) {
+        lua_pushfstring(L, "getTimestamp: bad argument #%d type (line number as number expected, got %s!", s, luaL_typename(L, s));
+        return lua_error(L);
     } else {
         luaLine = lua_tointeger(L, n);
+        if (luaLine < 1) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "line number %d invalid, it should be greater than zero", luaLine);
+            return 2;
+        }
     }
+
     Host& host = getHostFromLua(L);
-    if (name.empty()) {
+    if (name.isEmpty()) {
         if (luaLine > 0 && luaLine < host.mpConsole->buffer.timeBuffer.size()) {
```

Comment:
Yeah - I need guidance on this because I am not entirely sure I can do that preceding check because it prevents execution proceeding to the point that `"getTimestamp: invalid line number"` is returned rather than the traditional `nil` + `<error message>` if a zero or negative line is given.

Also - I am a little bit worried when I compare this one against `getLines(from, to)` which requires `from` to be `0` (and `to` be *greater* or *less* then `from`) in order to get the very ***first*** line in the `(QStringList) TBuffer::lineBuffer` - even though `0` based indexing is a C/C++ thing rather than the lua way.